src/nft/components/collection/TraitSelect.tsx

```diff
@@ -0,0 +1,191 @@
+import clsx from 'clsx'
+import useDebounce from 'hooks/useDebounce'
+import { pluralize } from 'nft/utils/roundAndPluralize'
+import { scrollToTop } from 'nft/utils/scrollToTop'
+import { FormEvent, MouseEvent } from 'react'
+import { useEffect, useLayoutEffect, useState } from 'react'
+
+import { subheadSmall } from '../../css/common.css'
+import { Trait, useCollectionFilters } from '../../hooks/useCollectionFilters'
+import { Box } from '../Box'
+import { Column, Row } from '../Flex'
+import { ChevronUpIcon } from '../icons'
+import { Checkbox } from '../layout/Checkbox'
+import * as styles from './Filters.css'
+
+const TraitItem = ({
+  trait,
+  addTrait,
+  removeTrait,
+  isTraitSelected,
+}: {
+  trait: Trait
+  addTrait: (trait: Trait) => void
+  removeTrait: (trait: Trait) => void
+  isTraitSelected: boolean
+}) => {
+  const [isCheckboxSelected, setCheckboxSelected] = useState(false)
+  const [hovered, setHovered] = useState(false)
+  const handleHover = () => setHovered(!hovered)
+  const toggleShowFullTraitName = useCollectionFilters((state) => state.toggleShowFullTraitName)
+
+  const { shouldShow, trait_value, trait_type } = useCollectionFilters((state) => state.showFullTraitName)
+  const isEllipsisActive = (e: MouseEvent<HTMLElement>) => {
+    if (e.currentTarget.offsetWidth < e.currentTarget.scrollWidth) {
+      toggleShowFullTraitName({
+        shouldShow: true,
+        trait_value: trait.trait_value,
+        trait_type: trait.trait_type,
+      })
+    }
+  }
+  useEffect(() => {
+    setCheckboxSelected(isTraitSelected)
+  }, [isTraitSelected])
+
+  const handleCheckbox = (e: FormEvent) => {
+    e.preventDefault()
+    scrollToTop()
+
+    if (!isCheckboxSelected) {
+      addTrait(trait)
+      setCheckboxSelected(true)
+    } else {
+      removeTrait(trait)
+      setCheckboxSelected(false)
+    }
+  }
+
+  const showFullTraitName = shouldShow && trait_type === trait.trait_type && trait_value === trait.trait_value
+
+  return (
+    <Row
+      key={trait.trait_value}
+      maxWidth="full"
+      overflowX={'hidden'}
+      overflowY={'hidden'}
+      fontWeight="normal"
+      className={`${subheadSmall} ${styles.subRowHover}`}
+      justifyContent="space-between"
+      cursor="pointer"
+      paddingLeft="12"
+      paddingRight="12"
+      style={{ paddingBottom: '21px', paddingTop: '21px', maxHeight: '44px' }}
+      onMouseEnter={handleHover}
+      onMouseLeave={handleHover}
+      onClick={handleCheckbox}
+    >
+      <Box
+        as="span"
+        whiteSpace="nowrap"
+        textOverflow="ellipsis"
+        overflow="hidden"
+        maxWidth={!showFullTraitName ? '160' : 'full'}
+        onMouseOver={(e) => isEllipsisActive(e)}
+        onMouseLeave={() => toggleShowFullTraitName({ shouldShow: false, trait_type: '', trait_value: '' })}
+      >
+        {trait.trait_type === 'Number of traits'
+          ? `${trait.trait_value} trait${pluralize(Number(trait.trait_value))}`
```

Comment:
can do `+` instead of casting to `Number`

Reply:
I think Number() is easier read or potentially parseInt().